mule-extensions-api/src/main/java/org/mule/runtime/extension/api/dsl/syntax/DslSyntaxUtils.java

```diff
@@ -94,8 +95,12 @@ static boolean isFlattened(ObjectFieldType field, MetadataType fieldValue) {
     return fieldValue instanceof ObjectType && field.getAnnotation(FlattenedTypeAnnotation.class).isPresent();
   }
 
-  static String getTypeKey(MetadataType type, String namespace, String namespaceUri) {
-    return getId(type).get() + namespace + namespaceUri;
+  static Optional<String> getTypeKey(MetadataType type, String namespace, String namespaceUri) {
+    Optional<String> id = getId(type);
+    if (id.isPresent()) {
+      return Optional.of(id.get() + namespace + namespaceUri);
+    }
+    return !"".equals(getAlias(type)) ? Optional.of(getAlias(type) + namespace + namespaceUri) : Optional.empty();
```

Comment:
Use org.mule.runtime.core.api.util.StringUtils#isEmpty instead of comparing against the empty string.